api/src/test/java/org/pentaho/metaverse/api/analyzer/kettle/ExternalResourceCacheTest.java

```diff
@@ -251,4 +237,58 @@ public void test_ExternalResourceExternalResourceValues() {
     // make sure we're getting a copy, not the original
     assertFalse( resourceValues.getInternal() == resourceValues.getInternal() );
   }
+
+  @Test
+  public void test_caching_timeout() throws Exception {
+    long timeoutInSeconds = 1;
+    IMetaverseConfig metaverseConfig = mock( IMetaverseConfig.class );
+    when( metaverseConfig.getExternalResourceCacheExpireTime() ).thenReturn( Long.toString( timeoutInSeconds ) );
+    testInstance = new ExternalResourceCache( metaverseConfig );
+
+    initMetas();
+
+    final ExternalResourceCache.ExternalResourceValues resources1 = testInstance
+      .newExternalResourceValues();
+    resources1.add( Mockito.mock( IExternalResourceInfo.class ) );
+
+    final ExternalResourceCache.ExternalResourceValues resources2 = testInstance
+      .newExternalResourceValues();
+    resources2.add( Mockito.mock( IExternalResourceInfo.class ) );
+
+    // verify nothing in cache
+    assertNull( testInstance.get( meta1 ) );
+    assertNull( testInstance.get( meta2 ) );
+
+    // add items
+    testInstance.cache( meta1, resources1 );
+    testInstance.cache( meta2, resources2 );
+    assertEquals( resources1, testInstance.get( meta1 ) );
+    assertEquals( resources2, testInstance.get( meta2 ) );
+
+    // sleep to test time eviction policy
+    Thread.sleep( 11 * timeoutInSeconds * 1000 / 10 ); // eviction policy 1.1  * timeout
```

Comment:
Going to commit code smell, no way around a thread sleep. Testing time eviction policy, wait time is < 2 seconds.
https://sonar.pentaho.net/project/issues?id=pentaho%3Apentaho-metaverse&issues=AXOGnWyW6uzjgHisa58V&open=AXOGnWyW6uzjgHisa58V&pullRequest=629